scipy/optimize/_differentialevolution.py

```diff
@@ -400,19 +425,23 @@ def __init__(self, func, bounds, args=(),
 
         # default population initialization is a latin hypercube design, but
         # there are other population initializations possible.
-        self.num_population_members = popsize * self.parameter_count
+        # the minimum is 5 because 'best2bin' requires a population that's at
+        # least 5 long
+        self.num_population_members = max(5, popsize * self.parameter_count)
```

Comment:
A minimum of 5 is enforced here but can be overridden if array initialisation is used with fewer than 5 members.
Suggest adding a check in ``init_population_array()`` that raises an exception if less than 5 members are passed.